File: mysql-jooq/src/main/java/by/next/way/jooq/java/MysqlConnector.java
package by.next.way.jooq.java;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.jooq.DSLContext;
import org.jooq.Record;
import org.jooq.SQLDialect;
import org.jooq.impl.DSL;

import java.sql.Connection;
import java.sql.DriverManager;
import java.util.List;

public class MysqlConnector {
    private Logger log = LogManager.getLogger();

    List<Record> execute(
            String sql,
            String url,
            String user,
            String password
    ) throws ClassNotFoundException {
        Class.forName("com.mysql.cj.jdbc.Driver");
        try (Connection connection = DriverManager.getConnection(url, user, password)) {
            DSLContext dslContext = DSL.using(connection, SQLDialect.MYSQL);
            return dslContext.fetchMany(sql).resultsOrRows().get(0).result();
        } catch (Exception e) {
            log.error("Jooq error!", e);
            return null;
        }
    }
}

File: mysql-jooq/src/main/kotlin/by/next/way/jooq/kotlin/MysqlConnector.kt
package by.next.way.jooq.kotlin

import org.apache.logging.log4j.LogManager
import org.jooq.Record
import org.jooq.SQLDialect
import org.jooq.impl.DSL
import java.sql.DriverManager

object MysqlConnector {

    private val log = LogManager.getLogger()

    fun execute(
            sql: String,
            url: String,
            user: String,
            password: String
    ): List<Record>? {
        Class.forName("com.mysql.cj.jdbc.Driver")
        try {
            DriverManager.getConnection(url, user, password).use { connection ->
                val dslContext = DSL.using(connection, SQLDialect.MYSQL)
                return dslContext.fetchMany(sql).resultsOrRows()[0].result().toList()
            }
        } catch (e: Exception) {
            log.error("Jooq error!", e)
            return null
        }
    }
}